### ISSUE TITLE
regexextract should have the option to extract a sub match (group)

### ISSUE THREAD

**Issue by gdoron** [metabase/metabase#33422]
@paoliniluis 
Currently there is no reasonable way of extracting the value doron from this:
```
'very_long_borying_text..and...who=doron;something-boring-continues'
```
One can use 
```
regexextract([column_name], "who=([a-zA-Z]+)")
```
But that would return `who=doron` while what I've actually wanted is something like:
`regexextract([column_name], "who=([a-zA-Z]+)", 1)` (return first matching **sub-group**)

**Comment by gdoron** [comment]
Actually I've found out it's more important than I thought.
Use the same function on a snowflake and a postgres database and you'll get different results. @paoliniluis

**Comment by paoliniluis** [comment]
Possibly because of different implementations by db engines ?

**Comment by gdoron** [comment]
Or different translation to SQL.
But anyway metabase should unify as much as possible, right?

**Comment by paoliniluis** [comment]
Agree, the transpilation should try to make it as agnostic as possible. I'll change the title to reflect it
